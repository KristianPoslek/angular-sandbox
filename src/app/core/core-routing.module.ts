import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [
    {
        path: '',
        loadChildren: '../pokemon-list/pokemon-list.module#PokemonListModule'
    },
    {
        path: 'pokemon',
        loadChildren:
            '../pokemon-details/pokemon-details.module#PokemonDetailsModule'
    }
];

@NgModule({
    imports: [RouterModule.forRoot(routes, { useHash: true })],
    exports: [RouterModule]
})
export class CoreRoutingModule {}
