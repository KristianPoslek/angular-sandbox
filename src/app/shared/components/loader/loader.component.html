<div *ngIf="loading > 0" class="pokeball" id="normal"></div>
