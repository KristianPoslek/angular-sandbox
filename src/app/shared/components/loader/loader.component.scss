$red: #e20f07;
$blk: #000;
$wht: #fff;
$gry: #a5a5a5;

@keyframes rotateBall {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}
@-webkit-keyframes rotateBall {
    0% {
        transform: rotate(0deg);
    }
    50% {
        transform: rotate(50deg);
    }
    0% {
        transform: rotate(360deg);
    }
}
.pokeball {
    width: 4px;
    height: 4px;
    transform-origin: 24px 24px;
    animation: rotateBall 1.5s infinite forwards;
    position: fixed;
    top: 15px;
    right: 60px;
    z-index: 10000;
    &#normal {
        box-shadow: 16px 0 0 $blk, 20px 0 0 $blk, 24px 0 0 $blk, 28px 0 0 $blk,
            //Row: 2
                8px 4px 0 $blk,
            12px 4px 0 $blk, 16px 4px 0 $red, 20px 4px 0 $red, 24px 4px 0 $red,
            28px 4px 0 $red, 32px 4px 0 $blk, 36px 4px 0 $blk,
            //Row: 3
                4px 8px 0 $blk,
            8px 8px 0 $red, 12px 8px 0 $red, 16px 8px 0 $wht, 20px 8px 0 $red,
            24px 8px 0 $red, 28px 8px 0 $red, 32px 8px 0 $red, 36px 8px 0 $red,
            40px 8px 0 $blk,
            //Row: 4
                4px 12px 0 $blk,
            8px 12px 0 $red, 12px 12px 0 $wht, 16px 12px 0 $wht,
            20px 12px 0 $wht, 24px 12px 0 $red, 28px 12px 0 $red,
            32px 12px 0 $red, 36px 12px 0 $red, 40px 12px 0 $blk,
            //Row: 5
                0px 16px 0 $blk,
            4px 16px 0 $red, 8px 16px 0 $red, 12px 16px 0 $red, 16px 16px 0 $wht,
            20px 16px 0 $red, 24px 16px 0 $red, 28px 16px 0 $red,
            32px 16px 0 $red, 36px 16px 0 $red, 40px 16px 0 $red,
            44px 16px 0 $blk,
            //Row: 6
                0px 20px 0 $blk,
            4px 20px 0 $red, 8px 20px 0 $red, 12px 20px 0 $red, 16px 20px 0 $red,
            20px 20px 0 $blk, 24px 20px 0 $blk, 28px 20px 0 $red,
            32px 20px 0 $red, 36px 20px 0 $red, 40px 20px 0 $red,
            44px 20px 0 $blk,
            //Row: 7
                0px 24px 0 $blk,
            4px 24px 0 $blk, 8px 24px 0 $red, 12px 24px 0 $red, 16px 24px 0 $blk,
            20px 24px 0 $wht, 24px 24px 0 $gry, 28px 24px 0 $blk,
            32px 24px 0 $red, 36px 24px 0 $red, 40px 24px 0 $blk,
            44px 24px 0 $blk,
            //Row: 8
                0px 28px 0 $blk,
            4px 28px 0 $wht, 8px 28px 0 $blk, 12px 28px 0 $blk, 16px 28px 0 $blk,
            20px 28px 0 $gry, 24px 28px 0 $gry, 28px 28px 0 $blk,
            32px 28px 0 $blk, 36px 28px 0 $blk, 40px 28px 0 $gry,
            44px 28px 0 $blk,
            //Row: 9
                4px 32px 0 $blk,
            8px 32px 0 $wht, 12px 32px 0 $wht, 16px 32px 0 $wht,
            20px 32px 0 $blk, 24px 32px 0 $blk, 28px 32px 0 $gry,
            32px 32px 0 $gry, 36px 32px 0 $gry, 40px 32px 0 $blk,
            //Row: 10
                4px 36px 0 $blk,
            8px 36px 0 $gry, 12px 36px 0 $wht, 16px 36px 0 $wht,
            20px 36px 0 $wht, 24px 36px 0 $gry, 28px 36px 0 $gry,
            32px 36px 0 $gry, 36px 36px 0 $gry, 40px 36px 0 $blk,
            //Row: 11
                8px 40px 0 $blk,
            12px 40px 0 $blk, 16px 40px 0 $gry, 20px 40px 0 $gry,
            24px 40px 0 $gry, 28px 40px 0 $gry, 32px 40px 0 $blk,
            36px 40px 0 $blk,
            //Row: 12
                16px 44px 0 $blk,
            20px 44px 0 $blk, 24px 44px 0 $blk, 28px 44px 0 $blk;
    }
}
