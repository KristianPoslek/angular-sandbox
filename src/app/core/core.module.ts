import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { PokemonDetailsModule } from '../pokemon-details/pokemon-details.module';
import { PokemonListModule } from '../pokemon-list/pokemon-list.module';
import { SharedModule } from '../shared/shared.module';
import { AppComponent } from './containers/app/app.component';
import { CoreRoutingModule } from './core-routing.module';
import { metaReducers } from './state/meta.reducers';

@NgModule({
    declarations: [AppComponent],
    imports: [
        BrowserModule,
        CoreRoutingModule,
        SharedModule.forRoot(),
        PokemonListModule,
        PokemonDetailsModule,
        StoreModule.forRoot({}, { metaReducers }),
        EffectsModule.forRoot([])
    ]
})
export class CoreModule {}
