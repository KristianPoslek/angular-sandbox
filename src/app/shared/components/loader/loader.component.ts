import { Component, OnDestroy, OnInit } from '@angular/core';
import { Subscription } from 'rxjs';
import { SharedStateService } from 'src/app/shared/state/services/shared-state.service';

@Component({
    selector: 'pkmn-loader',
    templateUrl: './loader.component.html',
    styleUrls: ['./loader.component.scss']
})
export class LoaderComponent implements OnInit, OnDestroy {
    loading: number;
    subscriptions: Subscription = new Subscription();

    constructor(private sharedStateService: SharedStateService) {}

    ngOnInit() {
        this.subscriptions.add(
            this.sharedStateService.getLayoutState().subscribe(data => {
                this.loading = data.loading;
            })
        );
    }

    ngOnDestroy() {
        this.subscriptions.unsubscribe();
    }
}
