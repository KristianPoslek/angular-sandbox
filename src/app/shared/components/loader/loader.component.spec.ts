import { Injector } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { EMPTY, Subject } from 'rxjs';
import { SharedStateService } from 'src/app/shared/state/services/shared-state.service';
import { MockSharedStateService } from 'src/app/shared/state/services/shared-state.service.mock';
import { LoaderComponent } from './loader.component';

describe('shared.LoaderComponent', () => {
    let testee: LoaderComponent;
    let fixture: ComponentFixture<LoaderComponent>;
    let fixtureInjector: Injector;
    let sharedStateServiceMock: SharedStateService;

    beforeEach(async () => {
        TestBed.configureTestingModule({
            declarations: [LoaderComponent],
            providers: [
                {
                    provide: SharedStateService,
                    useClass: MockSharedStateService
                }
            ]
        });

        fixture = TestBed.createComponent(LoaderComponent);
        testee = fixture.componentInstance;
        fixtureInjector = fixture.debugElement.injector;
        sharedStateServiceMock = fixtureInjector.get<SharedStateService>(
            SharedStateService
        );
    });

    describe('on component init', () => {
        beforeEach(() => {
            sharedStateServiceMock.getLayoutState = () => EMPTY;
        });

        it('should have a working snapshot', () => {
            testee.loading = 2;
            fixture.detectChanges();
            expect(fixture).toMatchSnapshot();
        });

        it('should subscribe to pokemon list data changes', done => {
            const subject = new Subject<any>();
            sharedStateServiceMock.getLayoutState = () => subject;
            testee.ngOnInit();

            expect(testee.loading).toBeUndefined();
            sharedStateServiceMock.getLayoutState().subscribe(() => {
                expect(testee.loading).toEqual(2);
                done();
            });

            subject.next({
                loading: 2
            });
        });
    });

    it('should perform on destroy actions', () => {
        testee.subscriptions.unsubscribe = jest.fn();
        testee.ngOnDestroy();

        expect(testee.subscriptions.unsubscribe).toHaveBeenCalled();
    });
});
