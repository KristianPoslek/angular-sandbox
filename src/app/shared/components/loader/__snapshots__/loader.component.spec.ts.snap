// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared.LoaderComponent on component init should have a working snapshot 1`] = `
<pkmn-loader
  loading={[Function Number]}
  sharedStateService={[Function MockSharedStateService]}
  subscriptions={[Function Subscription]}
>
  <div
    class="pokeball"
    id="normal"
  />
</pkmn-loader>
`;
